### PR TITLE
feat(dynamite_petstore_example): Init

### PR DESCRIPTION
**Pull request by provokateurin**
Depends on https://github.com/nextcloud/neon/pull/1136
https://github.com/nextcloud/neon/issues/761

### CODE REVIEW

**Comment by provokateurin** [comment]
@Leptopoda I can move the fix for the security schemes to a separate PR. With the applied patch it is not necessary anyway since it doesn't break this PR.

File: packages/dynamite/dynamite/lib/src/builder/client.dart [hunk 1]
```diff
@@ -498,8 +498,14 @@ final authentication = $client.authentications.firstWhereOrNull(
 
   yield* securityRequirements.map((final requirement) {
     final securityScheme = spec.components!.securitySchemes![requirement.keys.single]!;
+    final nameParts = <String?>[
+      securityScheme.type,
+      securityScheme.scheme,
+      securityScheme.$in,
+      securityScheme.name,
+    ].whereNotNull();
```
Comment (Leptopoda) [inline packages/dynamite/dynamite/lib/src/builder/client.dart:506]: I this fits better as a method in the security scheme object

**Review by Leptopoda — APPROVED** [review]
Please also open followup issues for:
- the authentication issue
- to test the petstore against the server

File: packages/dynamite/dynamite/lib/src/builder/client.dart (hunk 1 above)
Comment (provokateurin) [reply packages/dynamite/dynamite/lib/src/builder/client.dart:506]: I'll make a separate PR then

**Comment by provokateurin** [comment]
yq is failing because of different implementations... I'll find a solution for that

**Comment by provokateurin** [comment]
I'm now using the petstore specification from https://github.com/OAI/OpenAPI-Specification which doesn't have the authentication problems and doesn't need any patching or yq magic.

**Review by Leptopoda — APPROVED** [review]

**Comment by Leptopoda** [comment]
Be aware that this uses the utf8 codec so #1143 is conflicting.
I suggest landing the flutter upgrade first

**Pull request merged**